lib/core/presentation/pages/event/event_detail_page/guest_event_detail_page/sub_pages/guest_event_approval_status_page/guest_event_approval_status_page.dart

```diff
@@ -0,0 +1,174 @@
+
+    String title = '';
+    String description = '';
+    String buttonLabel = '';
+    SvgGenImage icon = Assets.icons.icApprovalPending;
+
+    if (isPendingJoinRequest) {
+      title = t.event.eventApproval.awaitingHostApprovalGuestViewTitle;
+      description =
+          t.event.eventApproval.awaitingHostApprovalGuestViewDescription;
+      buttonLabel = t.common.done;
+    }
+
+    if (isDeclinedJoinRequest) {
+      title = t.event.eventApproval.declinedApprovalGuestViewTitle;
+      description = t.event.eventApproval.declinedApprovalGuestViewDescription;
+      buttonLabel = t.common.done;
+      icon = Assets.icons.icApprovalDeclined;
+    }
+
+    if (isApproved) {
+      title = t.event.eventApproval.approvedApprovalGuestViewTitle;
+      description = t.event.eventApproval.approvedApprovalGuestViewDescription;
+      buttonLabel = t.event.eventApproval.completePayment;
+      icon = Assets.icons.icApprovalApproved;
+    }
+
+    return Scaffold(
+      appBar: const LemonAppBar(),
+      body: Column(
+        crossAxisAlignment: CrossAxisAlignment.center,
+        mainAxisSize: MainAxisSize.max,
+        children: [
+          const Spacer(),
+          _CircularAnimation(
+            icon: icon,
+          ),
+          const Spacer(),
+          SafeArea(
+            child: Padding(
+              padding: EdgeInsets.symmetric(horizontal: Spacing.smMedium),
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.center,
+                mainAxisSize: MainAxisSize.min,
+                children: [
+                  Text(
+                    title,
+                    style: Typo.extraLarge.copyWith(
+                      color: colorScheme.onPrimary,
+                      fontWeight: FontWeight.bold,
+                      fontFamily: FontFamily.nohemiVariable,
+                    ),
+                  ),
+                  SizedBox(height: Spacing.superExtraSmall),
+                  Text(
+                    description,
+                    textAlign: TextAlign.center,
+                  ),
+                  SizedBox(height: Spacing.xSmall * 4),
+                  if (isApproved)
+                    LinearGradientButton.primaryButton(
+                      onTap: () async {
+                        await AutoRouter.of(context).pop();
+                        AutoRouter.of(context).navigate(
+                          EventBuyTicketsRoute(event: event),
+                        );
+                      },
+                      label: buttonLabel,
+                    )
+                  else
+                    LinearGradientButton.secondaryButton(
+                      onTap: () => AutoRouter.of(context).pop(),
+                      label: buttonLabel,
+                    ),
+                ],
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _CircularAnimation extends StatelessWidget {
```

Comment:
Can you moved to common Widget about this animation, this reused quite a lot

Reply:
let raise to another PR, need to check which can be configurable if we make it common